src/UseCases/DisplayPage/PageContentModifier.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace WMDE\Fundraising\Frontend\UseCases\DisplayPage;
+
+use Psr\Log\LoggerInterface;
+
+/**
+ * @licence GNU GPL v2+
+ */
+class PageContentModifier {
+
+	private $logger;
+	private $substitutions;
+
+	public function __construct( LoggerInterface $logger, array $substitutions = [] ) {
+		$this->logger = $logger;
+		$this->substitutions = $substitutions;
+	}
+
+	public function getProcessedContent( string $content, string $pageName ): string {
+		# full HTML document usually indicates an error (e.g. access denied)
+		if ( preg_match( '/^<!DOCTYPE html/', $content ) ) {
+			$this->logger->debug( __METHOD__ . ': fail, got error page', [ $content ] );
+			return '';
+		}
+
+		# NOTE: don't strip comments, they may contain javascript, etc!
+		# $content = preg_replace('/<!--.*?-->/s', '', $content);
```

Comment:
They are copied yes. I've not really looked at them yet and do not know what their deal is.